### ISSUE TITLE
DnsNameResolverContext infinite loop

### ISSUE THREAD

**Issue by serphacker** [netty/netty#4771]
Got stuck there : 
https://github.com/netty/netty/blob/netty-4.1.0.Beta8/resolver-dns/src/main/java/io/netty/resolver/dns/DnsNameResolverContext.java#L280

I think I did receive a response to another query but with the same id.

```
"dns-3-1" #12 prio=5 os_prio=0 tid=0x00007f87c442d800 nid=0x1401 runnable [0x00007f87a9ca0000]
   java.lang.Thread.State: RUNNABLE
        at io.netty.resolver.dns.DnsNameResolverContext.onResponseCNAME(DnsNameResolverContext.java:288)
        at io.netty.resolver.dns.DnsNameResolverContext.onResponseAorAAAA(DnsNameResolverContext.java:264)
        at io.netty.resolver.dns.DnsNameResolverContext.onResponse(DnsNameResolverContext.java:164)
        at io.netty.resolver.dns.DnsNameResolverContext$2.operationComplete(DnsNameResolverContext.java:142)
        at io.netty.util.concurrent.DefaultPromise.notifyListener0(DefaultPromise.java:680)
        at io.netty.util.concurrent.DefaultPromise.notifyListeners(DefaultPromise.java:567)
        at io.netty.util.concurrent.DefaultPromise.setSuccess(DefaultPromise.java:397)
        at io.netty.resolver.dns.DnsQueryContext.setSuccess(DnsQueryContext.java:181)
        at io.netty.resolver.dns.DnsQueryContext.finish(DnsQueryContext.java:164)
        at io.netty.resolver.dns.DnsNameResolver$DnsResponseHandler.channelRead(DnsNameResolver.java:944)
        at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83)
        at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:163)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:155)
        at io.netty.handler.codec.MessageToMessageDecoder.channelRead(MessageToMessageDecoder.java:103)
        at io.netty.channel.ChannelHandlerInvokerUtil.invokeChannelReadNow(ChannelHandlerInvokerUtil.java:83)
        at io.netty.channel.DefaultChannelHandlerInvoker.invokeChannelRead(DefaultChannelHandlerInvoker.java:163)
        at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:155)
        at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:950)
        at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:94)
        at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:510)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:467)
        at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:381)
        at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:353)
        at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:742)
        at io.netty.util.concurrent.DefaultThreadFactory$DefaultRunnableDecorator.run(DefaultThreadFactory.java:137)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by normanmaurer** [comment]
@trustin PTAL ....

**Comment by slandelle** [comment]
I can help if @serphacker shares a reproducer.

**Comment by normanmaurer** [comment]
@serphacker ping

**Comment by trustin** [comment]
Should be fixed by #4946
